ext/strings_test.go

```diff
@@ -329,3 +330,229 @@ func TestStringsWithExtension(t *testing.T) {
+			expectedOutput: "percent on the right%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the left",
+			format:         "%%%s",
+			formatArgs:     `"percent on the left"`,
+			expectedOutput: "%percent on the left",
+		},
+		{
+			name:           "multiple substitutions",
+			format:         "%d %d %d, %s %s %s, %d %d %d, %s %s %s",
+			formatArgs:     `1, 2, 3, "A", "B", "C", 4, 5, 6, "D", "E", "F"`,
+			expectedOutput: "1 2 3, A B C, 4 5 6, D E F",
+		},
+		{
+			name:           "percent sign escape sequence support",
+			format:         "\u0025\u0025escaped \u0025s\u0025\u0025",
+			formatArgs:     `"percent"`,
+			expectedOutput: "%escaped percent%",
+		},
+		{
+			name:           "fixed point formatting clause",
+			format:         "%.3f",
+			formatArgs:     "1.2345",
+			expectedOutput: "1.234",
+		},
+		{
+			name:           "binary formatting clause",
+			format:         "this is 5 in binary: %b",
+			formatArgs:     "5",
+			expectedOutput: "this is 5 in binary: 101",
+		},
+		{
+			name:           "octal formatting clause",
+			format:         "%o",
+			formatArgs:     "11",
+			expectedOutput: "13",
+		},
+		{
+			name:           "lowercase hexadecimal formatting clause",
+			format:         "%h is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1e is 20 in hexadecimal",
+		},
+		{
+			name:           "uppercase hexadecimal formatting clause",
+			format:         "%H is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1E is 20 in hexadecimal",
+		},
+		{
+			name:           "string support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656c6c6f20776f726c6421",
+		},
+		{
+			name:           "lowercase scientific notation formatting clause",
+			format:         "%.6e",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f×\u202f10⁰³",
+		},
+		{
+			name:           "uppercase scientific notation formatting clause",
+			format:         "%.6E",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f×\u202f10⁰³",
+		},
+		{
+			name:           "locale support",
+			format:         "%.3f",
+			formatArgs:     "3.14",
+			locale:         "fr_FR",
+			expectedOutput: "3,140",
+		},
+		{
+			name:           "default precision for fixed-point clause",
+			format:         "%f",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280",
+		},
+		{
+			name:           "default precision for scientific notation",
+			format:         "%e",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280\u202f×\u202f10⁰⁰",
+		},
+		{
+			name:           "NaN support for fixed-point",
+			format:         "%f",
+			formatArgs:     `"NaN"`,
+			expectedOutput: "NaN",
+		},
+		{
+			name:           "positive infinity support for fixed-point",
+			format:         "%f",
+			formatArgs:     `"Infinity"`,
+			expectedOutput: "∞",
```

Comment:
Neat. Did go output this? Is this what we want? I'll admit "Inf" would be more mundane, but do we want to get into unicode output here?

Reply:
This [and this](https://github.com/google/cel-go/pull/617#discussion_r1095219394) are both due to how `text/message`'s `Sprintf` implementation works (see [here](https://go.dev/play/p/Q8P1gQTTh76) for a playground example).

I'm not particularly a fan of the Unicode output, but I'm also concerned that the alternative would be extremely difficult - building our own `%e` printing implementation to avoid Unicode would require us to handle `,` versus `.`, likely resulting in some kind of table similar to what `initTags` has for now.

Reply:
Ok, this is a bit fancier than I had hoped for, especially for "default" locale. Sadly I don't see anything in https://github.com/golang/text/blob/71a9c9afc4cd710b9412f7f99f0d8e35b10e488a/number/option.go that helps.  If no one else has better ideas I think I can live with this.

Reply:
(marking as resolved, seems everyone is fine with this)